target/linux/ramips/base-files/lib/upgrade/platform.sh

```diff
@@ -62,6 +62,7 @@ platform_check_image() {
 	gl-mt300n|\
 	gl-mt750|\
 	hc5*61|\
+	hc5962|\
```

Comment:
I'm quite sure the sysupgrade will fail at this point since your sysupgrade image doesn't start with the uImage magic 0x27051956.

The whole image header validation isn't required for the hc5962 since a validation based on the metadata will be done. Please add a new  block for the hc5962 containing:

```bash
hc5962)
	# these boards use metadata images
	return 0
	;;
```